Rubberduck.Core/UI/CodeExplorer/Commands/OpenDesignerCommand.cs

```diff
@@ -34,8 +38,26 @@ protected override bool EvaluateCanExecute(object parameter)
             try
             {
                 var declaration = node.Declaration;
-                return declaration != null && declaration.DeclarationType == DeclarationType.ClassModule &&
-                       _projectsProvider.Component(declaration.QualifiedName.QualifiedModuleName).HasDesigner;
+                if (declaration == null)
+                {
+                    return false;
+                }
+
+                var qualifiedModuleName = declaration.QualifiedName.QualifiedModuleName;
+
+                // ReSharper disable once SwitchStatementMissingSomeCases
+                switch (declaration.DeclarationType)
+                {
+                    case DeclarationType.ClassModule when qualifiedModuleName.ComponentType != ComponentType.Document:
+                        return _projectsProvider.Component(qualifiedModuleName).HasDesigner;
+                    case DeclarationType.ClassModule:
+                        using (var app = _vbe.HostApplication())
+                        {
+                            return app != null && app.CanOpenDocumentDesigner(qualifiedModuleName);
```

Comment:
So if we're using `app is null` because it's safer than `app == null`, should we also be using null propagation and null conditionals on statements like this?

`return app?.CanOpenDocumentDesigner(qualifiedModuleName) ?? false;`

Note - that's a question, not a request to change it.

Reply:
I agree this reads better. Updated.